pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -208,17 +208,17 @@ func (s *DiscoveryServer) ClearCacheFunc() func() {
 			// TODO: metric !!
 			return
 		}
+		s.env.PushContext = push
+		versionLocal := time.Now().Format(time.RFC3339)
 		initContextTime := time.Since(t0)
+		adsLog.Infof("InitContext %v for push takes %s", version, initContextTime)
 
 		// TODO: propagate K8S version and use it instead
 		versionMutex.Lock()
-		s.env.PushContext = push
-		version = time.Now().Format(time.RFC3339)
+		version = versionLocal
 		versionMutex.Unlock()
 
-		adsLog.Infof("Context init for push %v %s", initContextTime, version)
-
-		go s.AdsPushAll(versionInfo(), push)
+		go s.AdsPushAll(versionLocal, push)
```

Comment:
I think this version should be just the value set from upper, not set by other trigger when concurrent push.